src/availability/query_data.rs

```diff
@@ -429,3 +429,84 @@ pub(crate) fn payload_size<Types: NodeType>(payload: &Payload<Types>) -> u64 {
         Err(_) => 0,
     }
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "")]
+pub struct BlockSummaryQueryData<Types: NodeType> {
+    pub(crate) header: Header<Types>,
+    pub(crate) hash: BlockHash<Types>,
+    pub(crate) size: u64,
+    pub(crate) num_transactions: u64,
+    pub(crate) proposer_id: SignatureKey<Types>,
+}
+
+// Add some basic getters to the BlockSummaryQueryData type.
+impl<Types: NodeType> BlockSummaryQueryData<Types> {
+    pub fn header(&self) -> &Header<Types> {
+        &self.header
+    }
+
+    pub fn hash(&self) -> BlockHash<Types> {
+        self.hash
+    }
+
+    pub fn height(&self) -> u64 {
+        self.header.block_number()
+    }
+
+    pub fn size(&self) -> u64 {
+        self.size
+    }
+
+    pub fn num_transactions(&self) -> u64 {
+        self.num_transactions
+    }
+
+    pub fn proposer_id(&self) -> SignatureKey<Types> {
+        self.proposer_id.clone()
+    }
+}
+
+// Get the Proposer from the given header.  At the moment the proposer cannot
+// be determined from the header, however it is intended to in the future.
+// For now we will just generate one with dummy data.
+// TODO update this function to retrieve the proposer from the Header type once
+//      the Header type has been updated to include the proposer.
+pub fn get_proposer<Types: NodeType>(_header: &Header<Types>) -> SignatureKey<Types> {
```

Comment:
This should probably be private, or `pub(crate)` if necessary. It's not really part of the interface of this crate